### PR TITLE
Add the new attributes of releases endpoint

### PR DESCRIPTION
**Pull request by xiangge**
Add the new attributes of releases endpoint

### CODE REVIEW

File: spec/fixtures/vcr/works_with_where.yml [hunk 1]
```diff
@@ -19,31 +19,49 @@ http_interactions:
...
-      connection:
-      - close
       transfer-encoding:
       - chunked
       content-type:
       - application/json
     body:
       encoding: UTF-8
-      string: '{"count":0,"next":null,"previous":null,"results":[]}'
+      string: '{"count":25,"next":"https://pdc.host.dev.eng.pek2.redhat.com/rest_api/v1/releases/?active=false&page=2","previous":null,"results":[{"release_id":"rcmtools-1.5@rhel-7","short":"rcmtools","version":"1.5","name":"RCM
```
Comment (romanofski) [inline spec/fixtures/vcr/works_with_where.yml]: You sure you want to use these internal identifiers?

Comment (xiangge) [reply spec/fixtures/vcr/works_with_where.yml]: This is a dev env, and the same as previous cache address, so I think that's fine to use this.

**Comment by xiangge** [comment]
@romanofski as there are some columns of release endpoint just include in redhat pdc, so maybe we'd better not use the fedora PDC.

**Comment by romanofski** [comment]
@xiangge I hear you that it's a burden to change this, but I'm not very confident approving these patches which basically record internal traffic. This might be benign for this, but it basically puts a burden on every reviewer to carefully inspect the VCR records in order to not accidentally leak internal stuff. I'm not keen on having to have a talk with legal on this matter.

My preference would be:
a) change every internal identifier (hostnames, products, etc) to something which is public knowledge starting with this patch
b) have a convenient way to do this on recorded VCR sessions so anything which is currently in the repository is redacted

I suppose b) can be solved using with or without VCR, whatever is preferable.

**Comment by romanofski** [comment]
Ah well there is a third option now to think of it. You can also ignore my review obviously and just pick someone else who is fine to approve it.

**Comment by sthaha** [comment]
@xiangge I agree with @romanofski and the suggestions made. VCR does allow you to scrub off parts you don't want. Regarding internal information, is it feasible to record against the external / public facing pdc?

**Comment by xiangge** [comment]
@sthaha As there are some columns just in the internal pdc server like some releases endpoints' columns, so I think we can't use the public pdc server like fedora.
@romanofski  I agree with you that the data/url shouldn't include in the real internal data. Good point.

@sthaha and @romanofski  I can mock some data in local dev db without any internal real data or just modify the original data in files.

**Comment by simozhan** [comment]
@xiangge yeah. I agree with @romanofski for the upstream project, we'd better not leak the internal information. I think it is a good idea to setup a local dev db without any internal real data. The benefit is we don't need to change some existing cases as we only change the fields, not the amount of the resources.

**Comment by simozhan** [comment]
BTW. It seems you add allowed_push_targets for release/release_variants, do you want to also add allowed_push_targets for product and product_version. Or you want to submit it in another patch

**Comment by xiangge** [comment]
@simozhan Nice catch, I added the release_variants here is because that it is related to releases testcase.
and I will add this product related changes in another patch.

**Comment by simozhan** [comment]
@xiangge Ok. make sense to me.

**Comment by xiangge** [comment]
Change the test data back to the original ones and just add new column here, also change the url to pdc.test.com.
Please help to review again, thanks.

File: spec/fixtures/vcr/_page_count_returns_total_count.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 http_interactions:
 - request:
     method: get
-    uri: https://pdc.host.dev.eng.pek2.redhat.com/rest_api/v1/arches/?page=2
+    uri: https://pdc.test.com/rest_api/v1/arches/?page=2
```
Comment (romanofski) [inline spec/fixtures/vcr/_page_count_returns_total_count.yml]: Dear @xiangge, thanks for the update. Perhaps instead of `pdc.test.com` why not use the URI to Fedoras PDC. That's public knowledge and a deployed server. If the problem is that it might be confusing for other contributors I'd suggest to use an example domain like [.test](https://en.wikipedia.org/wiki/.test).

It would be good to keep just the URL changes in a separate patch as well.

**Comment by xiangge** [comment]
>  is this patch set now supposed to be free from internal identifiers?

@romanofski thanks for the reviewing, not sure if I am understanding your question, but this patch is based on the changes of feature parity requirements on pdc server side.

**Comment by romanofski** [comment]
Dear @xiangge, my review comment explained that I'm uncomfortable with the test data in use since most of them are internal identifiers. Even if some of them are public knowledge I want to have a process in place which redacts most of those identifiers to avoid mistakenly leaking company secrets. Unless this has been address I don't feel approving this patch. If you don't share the same concerns I'm happy if you guys dismiss the review and pick a different reviewer.

Note, I think @sthaha is perhaps not easily available to review this since he moved to a different team.

**Comment by xiangge** [comment]
@romanofski yes, I have noticed that @sthaha moved to another team.
You are the mainly reviewer/contributor, so your concern is important !  
And you reply so fast, that's nice, thanks:).

As I didn't change the original data here, just add some null columns, let's also see other guys suggestions about the current test data (@dacdownunder @simonbaird @simozhan ).

BTW: If we should do some changes, I think we'd better to give another patch to do this, like url and identifiers changes. One idea is to change the data to some fedora's and add some missing columns manually, but for me I have no idea if we need to this change.

**Comment by simonbaird** [comment]
So what is the concern about leaking internal details? Is it the hostname? Or is it the data in the VCR file that is considered to be sensitive?

**Comment by simonbaird** [comment]
Discussed it on IRC briefly. I think the question about internal data in the VCR files is out of scope for this patch. Since there is already internal data in the VCR files, I don't see the need to prevent these from being merged.

If we want to remove the real data, it can be addressed it in future patches.

**Comment by xiangge** [comment]
@simonbaird Yes, agree, thanks for the review.
